src/mediaembed.js

```diff
@@ -37,6 +44,89 @@ export default class MediaEmbed extends Plugin {
 	static get pluginName() {
 		return 'MediaEmbed';
 	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		this._attachAutoEmbedingEvents();
+	}
+
+	/**
+	 * Attach events required for "auto-embeding" feature.
+	 *
+	 * @private
+	 */
+	_attachAutoEmbedingEvents() {
+		const editor = this.editor;
+		const modelDocument = editor.model.document;
+		const mediaRegistry = editor.plugins.get( MediaEmbedEditing ).registry;
+
+		let leftLivePosition, rightLivePosition;
+
+		// We need to listen on `Clipboard#inputTransformation` because we need to save positions of selection.
+		// After pasting a content, between those position can be located a URL that should be transformed to media.
+		this.listenTo( editor.plugins.get( Clipboard ), 'inputTransformation', () => {
+			const firstRange = modelDocument.selection.getFirstRange();
+
+			leftLivePosition = LivePosition.createFromPosition( firstRange.start );
+			leftLivePosition.stickiness = 'toPrevious';
+
+			rightLivePosition = LivePosition.createFromPosition( firstRange.end );
+			rightLivePosition.stickiness = 'toNext';
+		} );
+
+		modelDocument.on( 'change:data', () => {
+			if ( !leftLivePosition ) {
+				return;
+			}
+
+			const urlRange = new Range( leftLivePosition, rightLivePosition );
+			const walker = new TreeWalker( { boundaries: urlRange, ignoreElementEnd: true } );
+
+			let url = '';
+
+			for ( const node of walker ) {
+				if ( node.type === 'elementStart' ) {
+					return detach();
+				}
+
+				url += node.item.data;
+			}
+
+			// If the url does not match to universal url regexp, let's skip that.
+			if ( !url.match( URL_REGEXP ) ) {
+				return detach();
+			}
+
+			// If the url is valid from MediaEmbed plugin point of view, let's use it.
+			if ( !mediaRegistry.hasMedia( url ) ) {
+				return detach();
+			}
+
+			// `leftLivePosition` won't be available in `setTimeout` function so let's clone it.
+			const positionToInsert = Position.createFromPosition( leftLivePosition );
+
+			global.window.setTimeout( () => {
```

Comment:
1. Paste.
2. Type.
3. Auto-embed after 500ms.
4. But the error is thrown. Probably because some ranges from 1. are no longer valid because of 2.

![kapture 2018-09-11 at 12 40 42](https://user-images.githubusercontent.com/1099479/45355198-124a2f00-b5c0-11e8-9c49-3930dfeb2f48.gif)

Reply:
Similarly

1. Paste `https://www.youtube.com/watch?v=H08tGjXNHO4`,
2. Undo,
3. Auto–embed after 500ms.

Also throws the `Uncaught CKEditorError: move-operation-nodes-do-not-exist: The nodes which should be moved do not exist.` error.

Reply:
That's the negative side of the timeout. I don't have an idea how we could solve it. Do you have any?